File: backend/internal/models/dto.go
package models

type CreateLeagueRequest struct {
	LeagueId   string `json:"leagueId"`
	LeagueName string `json:"leagueName"`
	TeamCount  string `json:"teamCount"`
}
type GetLeaguesIdsWithNameResponse struct {
	LeagueId   string `json:"leagueId"`
	LeagueName string `json:"leagueName"`
}

type GetActiveLeagueStandingsResponse struct {
	Standings []Standings `json:"standings"`
}
type GetFixturesResponse struct {
	UpcomingFixtures []Week `json:"upcomingFixtures"`
	PlayedFixtures   []Week `json:"playedFixtures"`
}

type EditMatchResult struct {
	LeagueId  string `json:"leagueId"`
	Home      string `json:"home"`
	Away      string `json:"away"`
	HomeScore int    `json:"homeScore"`
	AwayScore int    `json:"awayScore"`
	MatchWeek int    `json:"matchWeek"`
	Winner    string `json:"winner"`
}

type SimulateLeagueRequest struct {
	PlayAllFixture bool `json:"playAllFixture"`
}

type SimulationResponse struct {
	Matches          []Matches `json:"matches"`
	UpcomingFixtures []Week    `json:"upcomingFixtures"`
	PlayedFixtures   []Week    `json:"playedFixtures"`
}

File: frontend/src/interfaces/dto.ts
import type {Standings, Week} from "@/interfaces/league.ts";

export interface CreateLeagueRequest {
    leagueName: string;
    teamCount: string;
}

export interface GetLeaguesIdsWithNameResponse {
    leagueId: string;
    leagueName: string;
}

export interface GetActiveLeagueStandingsResponse {
    standings: Standings[];
}

export interface GetActiveLeagueFixturesResponse {
    upcomingFixtures: Week[];
    playedFixtures: Week[];
}

export interface UpdateStandingTableRequest {
    leagueId: string;
    teamName: string;
    points: number;
}

export interface SimulateLeagueRequest {
    playAllFixture: boolean;
}

export interface SimulationResponse {
    upcomingFixtures: Week[];
    playedFixtures: Week[];
}

export interface LeagueIdWithName {
    leagueId: string;
    leagueName: string;
}